hs/src/Reach/Compiler.hs

```diff
@@ -43,41 +43,31 @@ all_connectors =
       ]
 
 compile :: CompilerOpts -> IO ()
-compile copts = do
-  let outn = output copts
-  let outnMay outn_ = case intermediateFiles copts of
-        True -> Just outn_
-        False -> Nothing
+compile CompilerOpts {output, dirDotReach, intermediateFiles, installPkgs, source, tops} = do
+  let outnMay = flip doIf intermediateFiles
   let interOut outn_ = case outnMay outn_ of
         Just f -> LTIO.writeFile . f
         Nothing -> \_ _ -> return ()
-  dirDotReach' <- makeAbsolute $ dirDotReach copts
-  let doPkgs = installPkgs copts
-  djp <- gatherDeps_top (source copts) doPkgs dirDotReach'
+  dirDotReach' <- makeAbsolute dirDotReach
+
```

Comment:
No blank lines in do blocks (you did this in your other commit too and I neglected to complain)